data/data/bootstrap/baremetal/files/usr/local/bin/master-bmh-update.sh

```diff
@@ -15,9 +15,15 @@ while [ "$(oc get bmh -n openshift-machine-api -o name | wc -l)" -lt 1  ]; do
     sleep 20
 done
 
-while [ "$(oc get bmh -n openshift-machine-api -l installer.openshift.io/role=control-plane -o json | jq '.items[].status.provisioning.state' | grep -v provisioned -c)" -gt 0  ]; do
-    echo "Waiting for masters to become provisioned"
-    oc get bmh -A
+N="0"
+while [ "$N" -eq "0" ] ; do
+    N=$(oc get bmh -n openshift-machine-api -l installer.openshift.io/role=control-plane --no-headers=true | wc -l)
+    sleep 20
+done
+echo "Waiting for $N masters to become provisioned"
+while [ "$(oc get bmh -n openshift-machine-api -l installer.openshift.io/role=control-plane -o json | jq '.items[].status.provisioning.state' | grep provisioned -c)" -ne "$N"  ]; do
```

Comment:
I'm worried about a potential race where we catch Kubernetes with only 2 masters (3rd one just not created yet) and then keep waiting for exactly 2. Maybe change to `-ge`?

Reply:
Good point, (I guess you mean -lt ? )

I'll also move the sleep above so that it waits a little before trying to set the number

Reply:
Yeah, sorry, got the logic reversed in my head.